float32[] corner1
float32[] corner2
float32[] corner3
float32[] corner4
